src/dbnode/storage/series/buffer.go

```diff
@@ -537,7 +595,51 @@ func (b *dbBuffer) ReadEncoded(
 	return res, nil
 }
 
+func (b *dbBuffer) FetchBlocksForColdFlush(
+	ctx context.Context,
```

Comment:
This is what I meant by the `BlockingClose()` on the ctx probably being wrong since it will cause the data returned by this function to be finalized

Reply:
👍 fixed this.